### DISCUSSION TITLE
[Bug]: Academy: "Data Importer'" not accepting correct answers

### DISCUSSION THREAD

**Discussion by janlucaklees** [pimcore/pimcore#10730] [Q&A / Support] [answered]
### Expected behavior

On the 7th question of the Pimcore Juniour Certificate, "**What can be achieved with transformation pipelines and out-of-the-box operators?**", on the Data Importer exam. I choose these options:

- [x] Simple text transformations
- [x] Importing assets from remote locations
- [x] Load existing assets and data object based on ID, path or attributes.
- [ ] Transform data to number fields.
- [ ] Do complex statistical calculations
- [x] Assign data to object fields.

These are the answers I understand to be correct and the ones that were correct last year. 

I expect to have answered the question correctly.

### Actual behavior

No matter what options I choose, the 7th question is wrong.

(I tried probably all plausible combinations. Also this is my third or fourth week of trying.)

### Steps to reproduce

- Go to: https://pimcore.com/academy/course/Data-Importer~158351/exam
- Fill out the 7th question as I showed above.
- Fill out other questions, so you can submit the form.
- Question 7 will be wrong.

Here's my train of thought:
1. [x] Simple text transformations.
  This is shown in the material.
 
2. [x] Importing assets from remote locations.
  This is also shown / mentioned in the material.
 
3. [x] Load existing assets and data object based on ID, path or attributes.
  This is also shown in the material. It is talked about how you can load an object or assets to then assign it to whatever other object you are currently importing.

4. [ ] Transform data to number fields.
  What is this even supposed to mean? Data to fields? To me this sound like: "Can I cast a string to a variable?". So no, obviously I cannot take some imported data and turn it into a DAMN field. But I can cast imported data to a number, meaning a float or integer. This is also shown in the material. Either way, selecting this option or not, I cannot successfully answer this question.
  Also, this wasn't part of the accepted answer last year.
 
5. [ ] Do complex statistical calculations.
  Not mentioned in the material. And wasn't possible last year.
 
6. [x] Assign data to object fields.
  Obviously, this is the whole point, of the importer.

**Comment by dpfaffenbauer** [comment]
Right answer is: 
 - Transform data to number fields
 - Importing assets from remote locations
 - Simple text transformations
 - Load existing assets and data objects based on ID, path or attributes

**Reply by janlucaklees** [reply]
Thank you for the quick reply! I'm gonna try this out next week, when I can try again.

**Comment by BentBr** [comment, marked as answer]
Would you please explain why the last point "Assign data to object fields" is not one of the right answers?

As I understood one can load dataobjects and add / change data depending on the imported fields. Not?

**Reply by dpfaffenbauer** [reply]
the question is "What can be achieved with transformation pipelines and out-of-the-box operators?". so it asked you what "transformation" you can do, transformation pipelines are not about assigning data to the object, it is about actually changing the value before asigning it. trick question ;)

**Reply by BentBr** [reply]
🤦 
Thx, mate

**Reply by janlucaklees** [reply]
In that case I would highly recommend to rephrase the "Transform data to number fields" option. \
Not only does it not make sense. Data and fields are different things. One is actual information, and the other is a container for information. \
But it also is really unclear what you mean by that, especially with the "Assign data to object fields" option in there. Yes, I get that the word "transform" is different from "assign". But in generally are the options provided in your academy exams are rather hard to understand - better said, to interpret. In this context, I basically have to assume, that "Assign data to object fields" and "Transform data to number fields" kinda, probably, maybe mean the same thing. \

Also, I dislike that you are using trick-questions. They are not accessible. They make it harder for non-native or not so proficient speakers, heck, even I had problems and I would consider myself a rather good. And again, in the general context of your exam options being somewhat vague, they just make the academy frustrating. I don't learn about Pimcore so much that I learn how to interpret your questions.

So tl;dr: Make the options in your exam more specific.

**Reply by fashxp** [reply]
I changed it to 'Transform data to numeric values' ... should be more clear I guess. 

In general, we don't want to make the academy questions frustrating or trick questions. But we want to make sure, that certified developers understand certain things. 
Having that said: we are always open for feedback, especially when formulations are unclear or out of context. 

Thx for the feedback.

**Reply by dpfaffenbauer** [reply]
@janlucaklees I am not with Pimcore, so I don't make the questions. Therefore it is not MY academy ;). Just wanted to clarify that.

**Reply by janlucaklees** [reply]
@dpfaffenbauer Ahh, I see. I thought you were.

Thank you @fashxp for changing it!
